mkgu_packaging/dicarlo/marques/marques_stim_common.py

```diff
@@ -0,0 +1,243 @@
+        # apply operation to put a 1 for all points inclusively within the degree radius and a 0 outside it
+        self.mask = self.xv**2 + self.yv**2 <= rad ** 2
+
+    # same as circular mask but for an annulus
+    def set_annular_mask(self, inner_rad, outer_rad):
+        self.mask = (self.xv ** 2 + self.yv ** 2 <= outer_rad ** 2) * \
+                    (self.xv ** 2 + self.yv ** 2 > inner_rad ** 2)
+
+    def set_gaussian_envelope(self, sig):
+        d = np.sqrt(self.xv**2 + self.yv**2)
+        self.envelope = np.exp(-d**2/(2 * sig**2))
+        self.update_tex()
+
+    def show_stimulus(self):
+        # pyplot stuff
+        self.update_stimulus()
+        my_dpi = 192
+        fig = plt.figure()
+        fig.set_size_inches(self.size_px[1] / my_dpi, self.size_px[0] / my_dpi, forward=False)
+        ax = plt.axes([0, 0, 1, 1])
+        ax.set_axis_off()
+        fig.add_axes(ax)
+        ax.imshow(self.stimulus, cmap='gray')
+        plt.show()
+
+    def save_stimulus(self):
+        # save to correct (previously specified) directory
+        self.update_stimulus()
+        file_name = self.type_name + '_' + self.format_id.format(self.stim_id) + '.png'
+        imageio.imwrite(self.save_dir + os.sep + file_name, self.stimulus)
+        return file_name
+
+
+def load_stim_info(stim_name, data_dir):
+    stim = pd.read_csv(os.path.join(data_dir, 'stimulus_set'), dtype={'image_id': str})
+    image_paths = dict((key, value) for (key, value) in zip(stim['image_id'].values,
+                                                            [os.path.join(data_dir, image_name) for image_name
+                                                             in stim['image_file_name'].values]))
+    stim_set = StimulusSet(stim[stim.columns[:-1]])
+    stim_set.image_paths = image_paths
+    stim_set.identifier = stim_name
+
+    return stim_set
+
+
+def gen_blank_stim(degrees, size_px, save_dir):
+    if not (os.path.isdir(save_dir)):
+        os.mkdir(save_dir)
+
+    stim = Stimulus(size_px=[size_px, size_px], type_name='blank_stim', save_dir=save_dir, stim_id=0)
+
+    stimuli = pd.DataFrame({'image_id': str(0), 'degrees': [degrees]})
+    image_names = (stim.save_stimulus())
+
+    stimuli['image_file_name'] = pd.Series(image_names)
+    stimuli['image_current_local_file_path'] = pd.Series(save_dir + os.sep + image_names)
+
+    stimuli.to_csv(save_dir + os.sep + 'stimulus_set', index=False)
+
+
+def gen_grating_stim(degrees, size_px, stim_name, grat_contrast, grat_pos, grat_rad, grat_sf, grat_orientation,
+                     grat_phase, save_dir):
+    if not (os.path.isdir(save_dir)):
+        os.mkdir(save_dir)
+
+    width = degrees
+
+    nStim = len(grat_pos) * len(grat_pos) * len(grat_contrast) * len(grat_rad) * len(grat_sf) * len(grat_orientation) \
+            * len(grat_phase)
+
+    print('Generating stimulus: #', nStim)
+
+    stimuli = pd.DataFrame({'image_id': [str(n) for n in range(nStim)], 'degrees': [width] * nStim})
```

Comment:
I'm also confused by these degrees -- I thought you always need a reference for degrees to make sense, isn't it just pixels at this point?

Reply:
No because the grating is defined with properties in degrees (for example, spatial frequency or size). Yes, the image itself is in pixels, but the accompanying metadata only makes sense assuming that the image corresponds to a certain size in degrees. If it is not then the grating does not have the spatial frequency that is defined in the metadata for example.

Reply:
got it, thanks!